### ISSUE TITLE
Bug: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

### ISSUE THREAD

**Issue by dustydang-tyler** [typescript-eslint/typescript-eslint#9762]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-estree

### Repro Code

```TypeScript
Ran into an issue with the following errors:
```

### Expected Result

I expected that the package would have support for the latest typescript version

### Actual Result

![Screenshot of error when running lint.](https://github.com/user-attachments/assets/be16cae8-7e83-42d8-8cd5-1e8439864990)

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `8.0.1` |
| `@typescript-eslint/parser`            | `8.0.1` |
| `@typescript-eslint/rule-tester`       | `8.0.1` |
| `@typescript-eslint/scope-manager`     | `8.0.1` |
| `@typescript-eslint/typescript-estree` | `8.0.1` |
| `@typescript-eslint/type-utils`        | `8.0.1` |
| `@typescript-eslint/utils`             | `8.0.1` |
| `TypeScript`                           | `5.5.4` |
| `ESLint`                               | `8.57.0` |
| `node`                                 | `22.0.0` |

**Comment by bradzacher** [comment]
If you're seeing this then you have a very old version of our tooling.
https://typescript-eslint.io/troubleshooting/faqs/general#how-do-i-check-to-see-what-versions-are-installed

You need to deduplicate your dependencies so that you're using just one version to fix this.
